src/librustc_mir/transform/inline.rs

```diff
@@ -251,6 +255,19 @@ impl Inliner<'tcx> {
             }
         }
 
+        if self.use_simple_heuristic {
+            return callee_body.basic_blocks().len() == 1
+                && callee_body.basic_blocks()[BasicBlock::from_u32(0)]
+                    .statements
+                    .iter()
+                    .filter(|stmt| match stmt.kind {
+                        StatementKind::StorageLive(_) | StatementKind::StorageDead(_) => false,
+                        _ => true,
+                    })
+                    .count()
+                    < 4;
```

Comment:
This keeps iterating even when the count is above 4. This can be costly when the amount of statements in the entry block is very large.